diff --git a/android/build.gradle b/android/build.gradle
index 4bd1d66..1b90ce4 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -29,6 +29,10 @@ def reactNativeArchitectures() {
   return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
 }
 
+def isNewArchitectureEnabled() {
+    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+}
+
 def USE_HERMES = rootProject.ext.hermesEnabled
 
 repositories {
@@ -52,6 +56,7 @@ android {
 
   buildFeatures {
     prefab true
+    buildConfig true
   }
 
   
@@ -59,6 +64,7 @@ android {
   defaultConfig {
     minSdkVersion safeExtGet('minSdkVersion', 19)
     targetSdkVersion safeExtGet('targetSdkVersion', 31)
+    buildConfigField("boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString())
     var appProject = rootProject.allprojects.find {it.plugins.hasPlugin('com.android.application')}
     externalNativeBuild {
       cmake {
diff --git a/android/build/generated/source/buildConfig/release/com/ammarahmed/mmkv/BuildConfig.java b/android/build/generated/source/buildConfig/release/com/ammarahmed/mmkv/BuildConfig.java
new file mode 100644
index 0000000..a3813e5
--- /dev/null
+++ b/android/build/generated/source/buildConfig/release/com/ammarahmed/mmkv/BuildConfig.java
@@ -0,0 +1,10 @@
+/**
+ * Automatically generated file. DO NOT MODIFY
+ */
+package com.ammarahmed.mmkv;
+
+public final class BuildConfig {
+  public static final boolean DEBUG = false;
+  public static final String LIBRARY_PACKAGE_NAME = "com.ammarahmed.mmkv";
+  public static final String BUILD_TYPE = "release";
+}
diff --git a/android/src/main/java/com/ammarahmed/mmkv/RNMMKVJSIModulePackage.java b/android/src/main/java/com/ammarahmed/mmkv/RNMMKVJSIModulePackage.java
deleted file mode 100644
index 1cbb97e..0000000
--- a/android/src/main/java/com/ammarahmed/mmkv/RNMMKVJSIModulePackage.java
+++ /dev/null
@@ -1,15 +0,0 @@
-package com.ammarahmed.mmkv;
-
-import com.facebook.react.bridge.JSIModulePackage;
-import com.facebook.react.bridge.JSIModuleSpec;
-import com.facebook.react.bridge.JavaScriptContextHolder;
-import com.facebook.react.bridge.ReactApplicationContext;
-import java.util.Collections;
-import java.util.List;
-
-public class RNMMKVJSIModulePackage implements JSIModulePackage { 
-    @Override
-    public List<JSIModuleSpec> getJSIModules(ReactApplicationContext reactApplicationContext, JavaScriptContextHolder jsContext) {
-        return Collections.emptyList();
-    }
-}
\ No newline at end of file
diff --git a/android/src/main/java/com/ammarahmed/mmkv/RNMMKVPackage.java b/android/src/main/java/com/ammarahmed/mmkv/RNMMKVPackage.java
index e0b51d6..4f6481b 100644
--- a/android/src/main/java/com/ammarahmed/mmkv/RNMMKVPackage.java
+++ b/android/src/main/java/com/ammarahmed/mmkv/RNMMKVPackage.java
@@ -4,14 +4,59 @@ package com.ammarahmed.mmkv;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+import androidx.annotation.Nullable;
 
 import com.facebook.react.ReactPackage;
 import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.uimanager.ViewManager;
 import com.facebook.react.bridge.JavaScriptModule;
+import com.facebook.react.TurboReactPackage;
+import com.facebook.react.module.annotations.ReactModule;
+import com.facebook.react.module.model.ReactModuleInfo;
+import com.facebook.react.module.model.ReactModuleInfoProvider;
+
+public class RNMMKVPackage extends TurboReactPackage implements ReactPackage {
+
+  @Nullable
+  @Override
+  public NativeModule getModule(String name, ReactApplicationContext reactContext) {
+    return new RNMMKVModule(reactContext);
+  }
+
+  @Override
+  public ReactModuleInfoProvider getReactModuleInfoProvider() {
+    Class<? extends NativeModule>[] moduleList = new Class[] {
+        RNMMKVModule.class
+    };
+
+    final Map<String, ReactModuleInfo> reactModuleInfoMap = new HashMap<>();
+    for (Class<? extends NativeModule> moduleClass : moduleList) {
+      ReactModule reactModule = moduleClass.getAnnotation(ReactModule.class);
+
+      reactModuleInfoMap.put(
+          reactModule.name(),
+          new ReactModuleInfo(
+              reactModule.name(),
+              moduleClass.getName(),
+              true,
+              reactModule.needsEagerInit(),
+              reactModule.hasConstants(),
+              reactModule.isCxxModule(),
+              BuildConfig.IS_NEW_ARCHITECTURE_ENABLED));
+    }
+
+    return new ReactModuleInfoProvider() {
+      @Override
+      public Map<String, ReactModuleInfo> getReactModuleInfos() {
+        return reactModuleInfoMap;
+      }
+    };
+  }
 
-public class RNMMKVPackage implements ReactPackage {
   @Override
   public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
     return Arrays.<NativeModule>asList(new RNMMKVModule(reactContext));
diff --git a/ios/MMKVNative.mm b/ios/MMKVNative.mm
index 34a9449..c2d29be 100644
--- a/ios/MMKVNative.mm
+++ b/ios/MMKVNative.mm
@@ -7,7 +7,6 @@
 
 #import <React/RCTBridge+Private.h>
 #import <React/RCTUtils.h>
-#import <ReactCommon/RCTTurboModule.h>
 
 
 using namespace facebook;