diff --git a/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchAppOnDevice.js b/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchAppOnDevice.js
index cb69cae..54e59e6 100644
--- a/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchAppOnDevice.js
+++ b/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchAppOnDevice.js
@@ -38,7 +38,7 @@ function tryLaunchAppOnDevice(device, androidProject, adbPath, args) {
     applicationId
   } = androidProject;
   const applicationIdWithSuffix = [appId || applicationId, appIdSuffix].filter(Boolean).join('.');
-  const activityToLaunch = mainActivity.includes('.') ? mainActivity : [packageName, mainActivity].filter(Boolean).join('.');
+  const activityToLaunch = mainActivity.includes('.') ? [packageName, mainActivity].join('') : [packageName, mainActivity].filter(Boolean).join('.');
   try {
     // Here we're using the same flags as Android Studio to launch the app
     const adbArgs = ['shell', 'am', 'start', '-n', `${applicationIdWithSuffix}/${activityToLaunch}`, '-a', 'android.intent.action.MAIN', '-c', 'android.intent.category.LAUNCHER'];
