diff --git a/node_modules/@react-native-community/eslint-config/index.js b/node_modules/@react-native-community/eslint-config/index.js
index bb5e2e3..605cd15 100644
--- a/node_modules/@react-native-community/eslint-config/index.js
+++ b/node_modules/@react-native-community/eslint-config/index.js
@@ -73,7 +73,6 @@ module.exports = {
       ],
       env: {
         jest: true,
-        'jest/globals': true,
       },
       rules: {
         'react-native/no-inline-styles': 0,
